flink-clients/src/main/java/org/apache/flink/client/LocalExecutor.java

```diff
@@ -103,6 +110,9 @@ public void start() throws Exception {
 				
 				// create the embedded runtime
 				Configuration configuration = getConfiguration(this);
+				if(this.configuration != null) {
+					configuration.addAll(this.configuration);
+				}
```

Comment:
Wouldn't it be better to move the null check to the constructor and create the default `Configuration` there via a call to `getConfiguration`?